core/src/main/java/org/elasticsearch/monitor/jvm/JvmInfo.java

```diff
@@ -368,11 +375,7 @@ public void readFrom(StreamInput in) throws IOException {
         mem.readFrom(in);
         gcCollectors = in.readStringArray();
         memoryPools = in.readStringArray();
-        if (in.readBoolean()) {
-            usingCompressedOops = in.readBoolean();
-        } else {
-            usingCompressedOops = null;
-        }
+        useCompressedOops = in.readOptionalString();
```

Comment:
Not in master, good catch. I'll remove the optional for master but it will be needed on the 2.x backport.